Avt-4. Flight. Поиск городов по названию.

evnag opened the pull request:
Города из таблицы flight_airport вынесены в отдельную таблицу flight_city с полями country, timezone, name.
 Добавлена миграция для наполнения тестовыми данными таблиц flight_airport, flight_city
Добавлен эндпойнт для поиска города по названию или по подстроке названия  с длиной не меньше 3 символов. Поиск без учета регистра.

evnag committed:
Добавлены миграции - создание таблицы flight_city, удаление полей country, timezone, name из таблицы flight_airport, заполнение таблиц тестовыми данными

evnag committed:
Добавлен эндпойнт для поиска города по названию или по подстроке названия.

evnag committed:
Добавлены тесты CityService

RusioPavlovsky commented on src/test/java/com/aviatickets/flight/service/CityServiceTest.java:
```diff
@@ -0,0 +1,65 @@
+public class CityServiceTest {
+
+    @Mock
+    private CityRepository repository;
```
Здесь лучше не мокать репозиторий, а проверить с базой h2. Здесь значительная часть логики - это поиск в базе, хорошо бы это протестировать

RusioPavlovsky commented on src/main/java/com/aviatickets/flight/service/CityService.java:
```diff
@@ -0,0 +1,26 @@
+    private final CityRepository repository;
+
+    public ListResult<City> findCity(String name) {
+        if (name == null || name.length() < 3) {
```
Число лучше вынести в константу, так непонятно почему именно 3

evnag committed:
Таблицы flight_airport, flight_city связаны отношением многие-к-одому.
Добавлены миграции - изменение таблицы flight_airport, удаление старых тест данных, занесение новых данных

evnag committed:
Длина строки требуемая для поиска городов вынесена в константу.
Юнит тесты CityService изменены на интеграционные.

evnag committed:
Добавлен checkstyle.xml

Commit:
some fix
